### PR TITLE
scaling function and tests

### PR DESCRIPTION
**Pull request by alisiafadini**
Support anisotropic scaling function and unit tests

### CODE REVIEW

**Comment by tjlane** [comment]
@alisiafadini you tagged me as a "assignee" (to write the code) not a reviewer (to review the code) -- which is why I didn't get an email asking for a review! ;)

**Review by tjlane — COMMENTED** [review]
I will re-write the code a bit -- let's discuss the interface though.

Tests actually look pretty good, will review those in detail in a 2nd step once the interface has converged

File: meteor/scale.py [hunk 1]
```diff
@@ -1,15 +1,156 @@
...
 
-def scale_structure_factors(reference: rs.DataSet, dataset_to_scale: rs.DataSet) -> rs.DataSet:
-    """Apply an anisotropic scaling so that `dataset_to_scale` is on the same scale as `reference`.
+@overload
+def scale_structure_factors(
+    reference: rs.DataSeries,
+    dataset_to_scale: rs.DataSeries,
+    uncertainties: Optional[rs.DataSeries] = None,
+    inplace: Literal[False] = False,
+) -> Union[rs.DataSeries, tuple[rs.DataSeries, rs.DataSeries]]: ...
```
[the remark this answers is not in the document]
Comment (alisiafadini) [reply meteor/scale.py:23]: i got your comment about this in the old PR – I fixed it in another place but it slipped here!

File: meteor/scale.py [hunk 2]
```diff
@@ -1,15 +1,156 @@
...
-        C * exp{ -(h**2 B11 + k**2 B22 + l**2 B33 +
-                    2hk B12 + 2hl  B13 +  2kl B23) }
+    C * exp{ -(h**2 B11 + k**2 B22 + l**2 B33 +
+                2hk B12 + 2hl  B13 +  2kl B23) }
 
     This is the same procedure implemented by CCP4's SCALEIT.
 
+    Parameters:
+    reference (rs.DataSeries): Single-column DataSeries to use as the reference for scaling.
+    dataset_to_scale (rs.DataSeries): Single-column DataSeries to be scaled.
```
[the remark this answers is not in the document]
Comment (alisiafadini) [reply meteor/scale.py:42]: Well the DataSeries will give only one column right? That one will be scaled. But clearly this is confusing. We should discuss the interface I agree

File: meteor/scale.py [hunk 3]
```diff
@@ -1,15 +1,156 @@
...
+import numpy as np
 import reciprocalspaceship as rs
+import scipy.optimize as opt
+
+
+@overload
+def scale_structure_factors(
+    reference: rs.DataSeries,
+    dataset_to_scale: rs.DataSeries,
+    uncertainties: Optional[rs.DataSeries] = None,
```
[the remark this answers is not in the document]
Comment (alisiafadini) [reply meteor/scale.py:12]: possible you are confusing `rs.DataSet` and `rs.DataSeries`? See [here](https://rs-station.github.io/reciprocalspaceship/userguide/mtzdtypes.html)

File: meteor/scale.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (tjlane) [reply meteor/scale.py:42]: yep sorry - my mistake!

File: meteor/scale.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (tjlane) [reply meteor/scale.py:12]: yes sorry, my mistake!

**Pull request closed without merging**